lib/utils.js

```diff
@@ -1,7 +1,28 @@
 'use strict'
 
+function isDeflate (buffer) {
```

Comment:
what is the scope of this change?
Why should we copy-paste those modules into this one? (at least credits should be added)

Reply:
One less dependency that could be potentially compromised?

Reply:
isDeflate is now spec compliant and definetly a complete new product. I benched the performance of isDeflate an it has still the same performance with the replaced one.

I also added references to the corresponding RFCs to explain the magic numbers.